go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -538,11 +538,15 @@ func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node,
 	addRoute := func(prefix, nexthop string) error {
 		logicalRouterStaticRoute := nbdb.LogicalRouterStaticRoute{
 			ExternalIDs: map[string]string{
-				"ic-node": node.Name,
+				"ic-node":               node.Name,
+				types.NetworkExternalID: zic.GetNetworkName(),
 			},
 			Nexthop:  nexthop,
 			IPPrefix: prefix,
 		}
+		// Note that because logical router static route was created without types.NetworkExternalID external-ids,
```

Comment:
nit: 
```
// Note that because logical router static routes were originally created without types.NetworkExternalID external-ids,
// skip types.NetworkExternalID check in the predicate function to replace existing static route with correct
// external-ids on an upgrade scenario.
```

Reply:
fixed